internal/metrics/metrics.go

```diff
@@ -234,6 +234,57 @@ var (
 		},
 		[]string{"chain_name", "upstream_id", "url", "errorType"},
 	)
+
+	latencyStatus = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: metricsNamespace,
+			Subsystem: "healthcheck",
+			Name:      "latency",
+			Help:      "Latency of upstream.",
+		},
+		[]string{"chain_name", "upstream_id", "url"},
+	)
+
+	latencyStatusCheckRequests = promauto.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: metricsNamespace,
+			Subsystem: "healthcheck",
+			Name:      "latency_check_requests",
+			Help:      "Total latency check requests made.",
+		},
+		[]string{"chain_name", "upstream_id", "url"},
+	)
+
+	latencyStatusCheckDuration = promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: metricsNamespace,
+			Subsystem: "healthcheck",
+			Name:      "latency_check_duration_seconds",
+			Help:      "Latency of latency check requests.",
+			Buckets:   []float64{.005, .01, .025, .05, .1, .25, .5, 1, 2.5, 5, 10, 20, 40},
+		},
+		[]string{"chain_name", "upstream_id", "url"},
+	)
+
+	latencyStatusCheckErrors = promauto.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: metricsNamespace,
+			Subsystem: "healthcheck",
+			Name:      "latency_check_errors",
+			Help:      "Errors when checking latency of upstream.",
+		},
+		[]string{"chain_name", "upstream_id", "url", "errorType"},
+	)
+
+	latencyStatusHighLatencies = promauto.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: metricsNamespace,
+			Subsystem: "healthcheck",
+			Name:      "latency_check_high_latency",
+			Help:      "Latency of upstream too high.",
+		},
+		[]string{"chain_name", "upstream_id", "url", "errorType"},
```

Comment:
Can we include the JSON RPC method as a label to all these metrics you're adding?